source/includes/fundamentals/code-snippets/indexes/atlasVectorSearch.go

```diff
@@ -0,0 +1,161 @@
+		Similarity    string `bson:"similarity"`
+		Quantization  string `bson:"quantization"`
+	}
+
+	type vectorDefinition struct {
+		Fields []vectorDefinitionField `bson:"fields"`
+	}
+
+	// Sets the index name and type to "vectorSearch"
+	const indexName = "vector_index"
+	opts := options.SearchIndexes().SetName(indexName).SetType("vectorSearch")
+
+	// Defines the index definition
+	vectorSearchIndexModel := mongo.SearchIndexModel{
+		Definition: vectorDefinition{
+			Fields: []vectorDefinitionField{{
+				Type:          "vector",
+				Path:          "plot_embedding",
+				NumDimensions: 1536,
+				Similarity:    "dotProduct",
+				Quantization:  "scalar"}},
+		},
+		Options: opts,
+	}
+
+	// Creates the index
+	searchIndexName, err := coll.SearchIndexes().CreateOne(ctx, vectorSearchIndexModel)
+	if err != nil {
+		log.Fatalf("Failed to create the vector search index: %v", err)
+	}
+	// end-create-vector-search
+
+	// Creates an Atlas Search index
+	// start-create-atlas-search
+	// Sets the index name and type to "search"
+	const indexName = "atlas_search_index"
+	opts := options.SearchIndexes().SetName(indexName).SetType("search")
+
+	// Defines the index definition 
+	searchIndexModel := mongo.SearchIndexModel{
+		Definition: bson.D{
+			{Key: "mappings", Value: bson.D{
+				{Key: "dynamic", Value: false},
+				{Key: "fields", Value: bson.D{
+					{Key: "plot", Value: bson.D{
+						{Key: "type", Value: "string"},
+					}},
+				}},
+			}},
+		},
+		Options: opts,
+	}
+
+	// Creates the index
+	searchIndexName, err := coll.SearchIndexes().CreateOne(ctx, searchIndexModel)
+	if err != nil {
+		log.Fatalf("Failed to create the atlas search index: %v", err)
+	}
+	// end-create-atlas-search
+
+	// start-list-index
+	// Specify the index to retrieve
+	const indexName = "<indexName>"
+	opts := options.SearchIndexes().SetName(indexName)
+
+	// Retrieves the details of the specified index
+	cursor, err := coll.SearchIndexes().List(ctx, opts)
+
+	// Prints the index details to the console as JSON
+	var results []bson.D
+	if err := cursor.All(ctx, &results); err != nil {
+		log.Fatalf("Failed to unmarshal results to bson: %v", err)
+	}
+	res, err := json.Marshal(results)
+	if err != nil {
+		log.Fatalf("Failed to marshal results to json: %v", err)
+	}
+	fmt.Println(res)
+	// end-list-index
+
+	// start-update-index
+	// Specify the index name and the new index definition 
+	const indexName = "<indexName>"
+
+	type vectorDefinitionField struct {
+		Type          string `bson:"type"`
+		Path          string `bson:"path"`
+		NumDimensions int    `bson:"numDimensions"`
+		Similarity    string `bson:"similarity"`
+	}
+
+	type vectorDefinition struct {
+		Fields []vectorDefinitionField `bson:"fields"`
+	}
+	
+	definition := vectorDefinition{
+		Fields: []vectorDefinitionField{{
+			Type:          "vector",
+			Path:          "<fieldToIndex>",
+			NumDimensions: <numberOfDimensions>,
+			Similarity:    "dotProduct"}},
```

Comment:
This is kind of hard to read.
```suggestion
		Fields: []vectorDefinitionField{
			{
				Type:          "vector",
				Path:          "<fieldToIndex>",
				NumDimensions: <numberOfDimensions>,
				Similarity:    "dotProduct",
			}
		},
```